features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/WebPImageCompression.kt

```diff
@@ -25,8 +25,17 @@ internal class WebPImageCompression : ImageCompression {
         // preallocate stream size
         val byteArrayOutputStream = ByteArrayOutputStream(bitmap.allocationByteCount)
         val imageFormat = getImageCompressionFormat()
-        @Suppress("UnsafeThirdPartyFunctionCall") // stream is not null and image quality is between 0 and 100
-        bitmap.compress(imageFormat, IMAGE_QUALITY, byteArrayOutputStream)
+
+        @Suppress("SwallowedException")
+        try {
+            // stream is not null and image quality is between 0 and 100
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            bitmap.compress(imageFormat, IMAGE_QUALITY, byteArrayOutputStream)
+        } catch (e: IllegalStateException) {
+            // if the bitmap was recycled while we were working on it
```

Comment:
The bitmaps are recycled by the system based on the view lifecycle, and I can reproduce it fairly easily. I can add a fallback in base64serializer that creates the bitmap from the drawable in the case of an empty bytearray, so that we don't lose the bitmap entirely, at the cost of some added complexity to the flow